sdk/identity/azure-identity/test/ut/managed_identity_credential_test.cpp

```diff
@@ -3038,4 +3038,159 @@ namespace Azure { namespace Identity { namespace Test {
     EXPECT_LE(response2.AccessToken.ExpiresOn, response2.LatestExpiration + 3600s);
   }
 
+  TEST(ManagedIdentityCredential, ImdsCustomHost)
```

Comment:
Test code is largely taken from https://github.com/Azure/azure-sdk-for-cpp/pull/5997/files#diff-197459028eebd28ba2c99a46cb2749466ddbf715169e09328ae9102aa9885470L2910